app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -106,63 +116,111 @@ async function addEthereumChainHandler(
         },
       });
 
-      networkClientId = await upsertNetworkConfiguration(
-        {
+      if (existingNetwork) {
+        // A network for this chain id already exists.
+        // Update it with any new information.
+
+        const clonedNetwork = { ...existingNetwork };
```

Comment:
The spread operator only goes one level deep when copying the object, everything else will be shallow copied. This means that only the top level properties are copied by value. Any nested properties are copied by reference.
Besides the developer expectations when reading `cloned`, this is relevant here because when changing `clonedNetwork.rpcEndpoints` or `clonedNetwork.blockExplorerUrls` you are actually also changing `existingNetwork.rpcEndpoints` and `existingNetwork.blockExplorerUrls`.
However, because we are using Immer to within the controllers, that might be preventing any side effect here. It would still be a good practice to either update the `clonedNetwork` name or actually deepclone `existingNetwork` (for example lodash has a method for it)

Reply:
Right.  You can't _actually_ edit `existingNetwork` here, because it'll throw `Cannot assign to read only property 'rpcEndpoints' of object...`.  That's why it does more spreads on the arrays below instead of pushing.  I'll look into deep cloning it to avoid dev confusion.

Reply:
I made it use `cloneDeep` in https://github.com/MetaMask/metamask-extension/pull/26433/commits/8648ae0c12b2b9c728eddc239b10837748541111